lms/djangoapps/commerce/utils.py

```diff
@@ -109,3 +117,263 @@ def upgrade_url(self, user, course_key):
             else:
                 return reverse('verify_student_upgrade_and_verify', args=(course_key,))
         return None
+
+
+def refund_entitlement(course_entitlement):
```

Comment:
Are there tests for this method?

Reply:
No this was the testing hole I mentioned earlier that I plan to test manually for now.  I am looking at the tests I would want for this now.

Reply:
My next post will have these tests.